File: lua/p10.lua
#!/usr/bin/lua

function TableLength(T)
  local count = 0
  for _ in pairs(T) do count = count + 1 end
  return count
end

limit = 2000000
if(TableLength(arg) > 2) then
  limit = arg[1]
end
cross_limit = math.floor(math.sqrt(limit))
sieve = {}
for i=0,limit do sieve[i] = false end
for n=4,limit,2 do
  sieve[n] = true
end
for n=3,cross_limit,2 do
  if(not sieve[n]) then
    for m=n*n,limit,2*n do
      sieve[m]=true
    end
  end
end

sum = 0
for n=2,limit do
  if (not sieve[n]) then
    sum = sum + n
  end
end

print(sum)

File: r/p10.r
#!/usr/bin/Rscript

limit <- 2000000
args <- commandArgs(TRUE)
if(length(args)>0){
  limit <- as.integer(args[1])
}
crosslimit <- floor(sqrt(limit))
sieve <- rep(F,2000000)
for(n in seq(4, limit, 2)){
  sieve[n] <- T
}
for (n in seq(3, crosslimit, 2)){
  if (! sieve[n]){
    for (m in seq(n*n, limit, 2*n)){
      sieve[m] <-  T
    }
  }
}
sum <- 0
for (n in 2:limit){
  if(! sieve[n]){
    sum <- sum + n
  }
}
print(sum)
